src/headless/blog/services/blog-feed-service.ts

```diff
@@ -0,0 +1,274 @@
+import { categories, posts, tags } from '@wix/blog';
+import { members } from '@wix/members';
+import { media } from '@wix/sdk';
+import {
+  defineService,
+  implementService,
+  type ServiceAPI,
+} from '@wix/services-definitions';
+import type { Signal } from '@wix/services-definitions/core-services/signals';
+import { SignalsServiceDefinition } from '@wix/services-definitions/core-services/signals';
+
+export interface PostResolvedFields {
+  owner: members.Member | null | undefined;
+  categories: posts.Category[];
+  tags: tags.BlogTag[];
+  coverImageUrl: string | null;
+  coverImageAlt: string | null;
+}
+
+export interface PostWithResolvedFields extends posts.Post {
+  resolvedFields: PostResolvedFields;
+}
+
+export const BlogFeedServiceDefinition = defineService<{
+  posts: Signal<PostWithResolvedFields[]>;
+  category: Signal<categories.Category | null>;
+  isLoading: Signal<boolean>;
+  error: Signal<string | null>;
+  totalPosts: number;
+  isEmpty: () => boolean;
+  hasNextPage: () => boolean;
+  loadNextPage: () => Promise<void>;
+}>('blogService');
+
+export type BlogFeedServiceAPI = ServiceAPI<typeof BlogFeedServiceDefinition>;
+
+export type BlogFeedServiceConfig = {
+  initialPosts?: PostWithResolvedFields[];
+  initialCategory?: categories.Category;
+  totalPostCount?: number;
+  pageSize?: number;
+  nextPageCursor?: string;
+};
+
+export const BlogFeedService =
+  implementService.withConfig<BlogFeedServiceConfig>()(
+    BlogFeedServiceDefinition,
+    ({ getService, config }) => {
+      const signalsService = getService(SignalsServiceDefinition);
+
+      const postsSignal = signalsService.signal<PostWithResolvedFields[]>(
+        config.initialPosts || []
+      );
+      const categorySignal = signalsService.signal<categories.Category | null>(
+        config.initialCategory || null
+      );
+      const isLoadingSignal = signalsService.signal<boolean>(false);
+      const errorSignal = signalsService.signal<string | null>(null);
+
+      let nextPageCursor: string | undefined = config.nextPageCursor;
+
+      const totalPosts = config.totalPostCount || 0;
+
+      const isEmpty = (): boolean => {
+        return !isLoadingSignal.get() && totalPosts === 0;
+      };
+
+      const hasNextPage = (): boolean => {
+        return !!nextPageCursor;
+      };
+
+      // Actions
+      const loadNextPage = async (): Promise<void> => {
+        try {
+          if (!nextPageCursor) {
+            return;
+          }
+
+          isLoadingSignal.set(true);
+          errorSignal.set(null);
+
+          let query = posts
```

Comment:
same comment about using platformized sort and filter over building it in the component